asyncapi-cli/src/main/java/io/ballerina/asyncapi/codegenerator/application/CodeGenerator.java

```diff
@@ -109,4 +118,11 @@ private String getFileContent(FileRepository fileRepository, String specPath) th
             throw new BallerinaAsyncApiException("Unknown file type: ".concat(specPath));
         }
     }
+
+    private Boolean isEventIdentifierInBody(String eventIdentifierPath) {
+        if (eventIdentifierPath.startsWith(Constants.CLONE_WITH_TYPE_VAR_NAME)) {
```

Comment:
Shall we check whether the type field of x-ballerina-event-identifier[1] is body and determine this?
[1]. https://github.com/ballerina-platform/asyncapi-triggers/blob/main/asyncapi/slack/asyncapi.yml#L3

Reply:
This change is done in [b6ae9c1](https://github.com/ballerina-platform/asyncapi-tools/pull/54/commits/b6ae9c1a6a84c715f44126e6cd85ee493177a302)